torchvision/datasets/imagenet.py

```diff
@@ -0,0 +1,229 @@
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        loader (callable, optional): A function to load an image given its path.
+
+     Attributes:
+        classes (list): List of the class names.
+        class_to_idx (dict): Dict with items (class_name, class_index).
+        wnids (list): List of the WordNet IDs.
+        class_to_idx (dict): Dict with items (wordnet_id, wordnet_id_index).
+        imgs (list): List of (image path, class_index) tuples
+        targets (list): The class_index value for each image in the dataset
+    """
+
+    def __init__(self, root, split='train', download=False, **kwargs):
+
+        root = self.root = os.path.expanduser(root)
+        self.split = self._verify_split(split)
+
+        if download:
+            self.download()
+
+        self.wnids, self.wnid_to_idx, classes, class_to_idx = self._load_meta()
+        super(ImageNet, self).__init__(self.split_folder, **kwargs)
+        self.root = root
+        self.classes = classes
+        self.class_to_idx = class_to_idx
+
+    def download(self):
+        self._empty_split_folder()
+
+        meta_file = os.path.join(self.root, META_DICT['filename'])
+        if not check_integrity(meta_file, META_DICT['md5']):
+            tmpdir = os.path.join(self.root, 'tmp')
+
+            archive_dict = ARCHIVE_DICT['devkit']
+            download_and_extract_tar(archive_dict['url'], self.root,
+                                     extract_root=tmpdir,
+                                     md5=archive_dict['md5'])
+            devkit_folder = _splitexts(os.path.basename(archive_dict['url']))[0]
+            meta, val_wnids = parse_devkit(os.path.join(tmpdir, devkit_folder))
+            torch.save((meta, val_wnids), meta_file)
+
+            shutil.rmtree(tmpdir)
+
+        archive_dict = ARCHIVE_DICT[self.split]
+        download_and_extract_tar(archive_dict['url'], self.root,
+                                 extract_root=self.split_folder,
+                                 md5=archive_dict['md5'])
+
+        if self.split == 'train':
+            prepare_train_folder(self.split_folder)
+        elif self.split == 'val':
+            val_wnids = torch.load(meta_file)[1]
+            prepare_val_folder(self.split_folder, val_wnids)
+
+    def _load_meta(self):
+        # TODO: verify meta file
+        return torch.load(os.path.join(self.root, META_DICT['filename']))[0]
+
+    def _empty_split_folder(self):
+        try:
+            shutil.rmtree(self.split_folder)
+        except FileNotFoundError:
```

Comment:
This is Python3 only unfortunately.

Reply:
I've put in 
```python
if sys.version_info[0] == 2:
    # FIXME: I don't know if this is good pratice / robust
    FileNotFoundError = OSError
```
on top of the module. This worked for me in a quick test, but the `FIXME` note should be taken literally.